toolkit/tools/imagecustomizerapi/overlay.go

```diff
@@ -5,69 +5,86 @@ package imagecustomizerapi
 
 import (
 	"fmt"
+	"path"
 	"strings"
 )
 
 type Overlay struct {
-	LowerDir  string               `yaml:"lowerDir"`
-	UpperDir  string               `yaml:"upperDir"`
-	WorkDir   string               `yaml:"workDir"`
-	Partition *IdentifiedPartition `yaml:"partition"`
+	LowerDirs         []string `yaml:"lowerDirs"`
+	UpperDir          string   `yaml:"upperDir"`
+	WorkDir           string   `yaml:"workDir"`
+	MountPoint        string   `yaml:"mountPoint"`
+	IsRootfsOverlay   bool     `yaml:"isRootfsOverlay"`
+	MountDependencies []string `yaml:"mountDependencies"`
+	MountOptions      string   `yaml:"mountOptions"`
 }
 
 func (o *Overlay) IsValid() error {
-	// Validate paths for UpperDir, WorkDir, and LowerDir
+	// Validate paths for LowerDirs, UpperDir, WorkDir, and MountPoint.
+	for _, lowerDir := range o.LowerDirs {
+		if err := validatePath(lowerDir); err != nil {
+			return fmt.Errorf("invalid lowerDir (%s):\n%w", lowerDir, err)
+		}
+	}
 	if err := validatePath(o.UpperDir); err != nil {
 		return fmt.Errorf("invalid upperDir (%s):\n%w", o.UpperDir, err)
 	}
 	if err := validatePath(o.WorkDir); err != nil {
 		return fmt.Errorf("invalid workDir (%s):\n%w", o.WorkDir, err)
 	}
-	if err := validatePath(o.LowerDir); err != nil {
-		return fmt.Errorf("invalid lowerDir (%s):\n%w", o.LowerDir, err)
+	if err := validatePath(o.MountPoint); err != nil {
+		return fmt.Errorf("invalid mountPoint (%s):\n%w", o.MountPoint, err)
+	}
+	for _, dependency := range o.MountDependencies {
+		if err := validatePath(dependency); err != nil {
+			return fmt.Errorf("invalid mountDependencies (%s):\n%w", dependency, err)
+		}
+	}
+
+	if strings.Contains(o.MountOptions, " ") {
```

Comment:
It would probably be good to also exclude tabs and newlines.

Also, if you could apply the same check to `MountPoint.Options` that would be greatly appreciated.